projects/hathor-wallet-headless/0001-multisig-create-mint-melt-token.md

```diff
@@ -0,0 +1,324 @@
+
+The data fields are:
+* `token`*
+* `amount`*
+* `deposit_address`
+* `change_address`
+* `melt_authority_address`
+* `allow_external_melt_authority_address`
+
+*`*` as required field 
+
+The response scheme for success:
+```ts
+{
+  "success": boolean,
+  "txHex": string
+}
+```
+
+The purpose of the command is only produce the `txHex` not signed and nothing more, it will not push the transaction to the network. The returned `txHex` will be handled by the coordinator service, and later on by the federation service, which is responsible to sign the transaction.
+
+With this specification we cover the possibility to melt tokens, send it to another wallet, create or not another melt authority, send the authority to another wallet, and define an address to deposit the withdraw amount if any.
+
+## Decode transaction with wallet metadata
+
+There is an endpoint to [inspect the transaction](https://hathor.gitbook.io/hathor/guides/multisig-wallet/sending-transactions#5.-check-the-multisig-transaction) by decoding the transaction hexadecimal encode in `POST:/wallet/decode`. However, it doesn't bring signature information, nor wallet information with it.
+
+By extending this endpoint operation to include wallet metadata we introduce a breaking change in the API. However, we keep coherence, once it is used not only to decode a `txHex` but to decode a partial transaction. The extension adds a semantic decoding of the transaction components regarding the wallet. Also, it reduces the roundtrips to the wallet by delivering the tokens balance.
+
+The data field is:
+* txHex
+* partial_tx
+
+The response scheme for success:
+```ts
+{
+  "success": boolean,
+  "complete": boolean,
+  "tx": {
+    "version": number,
+    "tokens": string[],
+    "inputs": [
+      {
+	    txId: string,
+        index: number,
+        decoded: {
+            type: string,
+            address: string,
+            timelock: number,
+        },  // data from the spent ouput
+        token: string,
+        value: number,
+        tokenData: number, // user face
+        token_data: number, // internal use
+        script: string,
+        signed: boolean,
+        mine: boolean,
+      },
+    ],
+    "outputs": [
+      {
+	    decoded: {
+		    address: string,
+		    timelock: number,
+		    mine: boolean,
+	    },
+        value: number,
+        tokenData: number, // user face
+        token_data: number, // internal use
+        script: string,
+        type: string,
+        token?: string,
+      }
+    ],
+  },
+  "balance": {
+    "<token-uid>":  {
+      "tokens": { available: number, locked: number ),
+      "authorities": {
+        mint: { available: number, locked: number ),
+        melt: { available: number, locked: number ),
+      },
+    },
+  },
+}
+```
+
+Lets represent the response document as ` $ `, and each element of a list ` [*] `, we have:
+* `$.complete` -- that represents the completeness of signatures required to use the transaction
```

Comment:
Should we go with something that makes more sense? Like `has_all_signatures`. Or do we use this term (`complete`) for this same context already in other places?

Reply:
This is the same as checking all `$.tx.inputs[*].signed == true`, right?

Reply:
Oh! I forgot to update it here, now we call this attribute `completeSignatures`.